### ISSUE TITLE
Create site manual "Windows 10"

### ISSUE THREAD

**Issue by elliekeli** [HelloZeroNet/ZeroNet#1241]
Hi,

I found this tutorial but it does not work for me. (Manual way: Using the command line1. Create site structure): http://zeronet.readthedocs.io/en/latest/using_zeronet/create_new_site/

If I'm not wrong, there's a bug in the "/"
"../Python/python.exe" zeronet.py siteCreate

(That's right)
"..\Python\python.exe" zeronet.py siteCreate

I installed "python-2.7.14.msi"

and since I did not find the "ZeroBundle/ZeroNet" folder does not exist... (Windows build)

I followed this:
cd C:\Users\NAME\Desktop\ZeroNet-win-dist\core
"C:\Python27\python.exe" zeronet.py siteCreate

I will get Error:
https://ctrlv.cz/shots/2018/01/14/k7e5.png

Thx help me.

**Comment by imachug** [comment]
That's incorrect use of `ZeroNet-win-dist`. Instead of:

```
cd C:\Users\NAME\Desktop\ZeroNet-win-dist\core
"C:\Python27\python.exe" zeronet.py siteCreate
```

...use:

```
cd C:\Users\NAME\Desktop\ZeroNet-win-dist
zeronet.exe siteCreate
```

Also, I don't recommend you to use `siteCreate`. Instead, use ZeroHello.

**Comment by imachug** [comment]
> If I'm not wrong, there's a bug in the "/"
> "../Python/python.exe" zeronet.py siteCreate
> 
> (That's right)
> "..\Python\python.exe" zeronet.py siteCreate

In Windows both `/` and `\` are correct.

**Comment by HelloZeroNet** [comment]
actually it's `lib/zeronet.cmd siteCreate` as the zeronet.exe is gui-only application and you won't see any output to command line from it.

But you can also use the web interface to create new site. `⋮ > Create new, empty site` on ZeroHello

**Comment by elliekeli** [comment]
Thx, it works.

PS:
I would like to ask how exactly "Identity address" works...
https://ctrlv.cz/shots/2018/01/14/mOoJ.png
I did not find anything in the documentation to help me understand how it works.

**Comment by imachug** [comment]
@HelloZeroNet

> actually it's lib/zeronet.cmd siteCreate as the zeronet.exe is gui-only application and you won't see any output to command line from it.

Well, you are not quite right. I use hooks in Git Center, and for signing call either `zeronet.py siteSign` or `core/zeronet.py siteSign` or `zeronet.exe siteSign`. All work, including `zeronet.exe`.

@elliekeli 

> I would like to ask how exactly "Identity address" works...

In clearnet you have a login, a password and sometimes a nick. In ZeroNet, your nick would be `elliekeli`, your login would be `1AoDYdMxwA55FTp2FNBrMaMv9r1ZzC6jSu`, and password would be a private key (which is stored in `core/data/users.json`). ZeroNet shows that login in sidebar.

ZeroNet is decentralized, so there can be options when nickname isn't unique (e.g. KaffieID). But login is auto-generated and guaranteed to be unique. Password is also auto-generated, so there is no chance you would be hacked (if someone doesn't steal your `users.json`, of course).

That identity address was given to you when you registered on ZeroID.

Hope I could answer your question.

**Comment by elliekeli** [comment]
Thank you very much for your answer!

if I understand it correctly "Identity address" is the unique account (for each zeronet page separately) everyone PC has "random generated"?

I still want to ask what it is "(limit used: 0.00kB / 0.00kB)" thx.

Also excuse my bad English...

**Comment by HelloZeroNet** [comment]
> All work, including zeronet.exe.

Yes, it does work, but using the zeronet.exe you won't see any response/output/message in the command line.

**Comment by imachug** [comment]
@HelloZeroNet Okay, that makes sense. I did that `>/dev/null` stuff to supress output, so I didn't notice.

@elliekeli 

> if I understand it correctly "Identity address" is the unique account (for each zeronet page separately) everyone PC has "random generated" on the same page?

When you log in some site, you see options like `Unique to this site` and `zeroid.bit`. If `Unique to this site` is chosen then identity address is really unique. When `zeroid.bit` is chosen, then that identity address would be in fact your ZeroID login.

> I still want to ask what it is "(limit used: 0.00kB / 0.00kB)" thx.

Site owners can set limits for users, so users could't use a lot of space. On ZeroTalk, that limit is 50KB. If you open that sidebar on ZeroTalk, you'll see `?KB/50KB` which means you have `(50 - ?) KB` storage left for your posts.

Of course `0KB/0KB` doesn't make sense, that means that you didn't log in yet.

**Comment by JazzMaster** [comment]
first of all- should never be a .exe. 

Its a python file, not a binary. where did you DL this from? sounds malware-ish. 

Its a shell scrypt in unix, batch file in sin. Use of .run is inconsistent with python in main scrypt. indicates use of a binary - which is wrong. Zero is and always will be implemented in python- and you will need the interpreter called at some point. You shouldnt be launching bins and the only use case I can see is a self-extracting python version.

python Zeronet.sh or python zeronet.py

unices have the advantage of using hashbomb commands to internally invoke the interpreter when accessing the file in execution.

NO, the / and \ are different. You must be in a advanced shell of some sort, not the command console. DOS takes one style and unix takes the other.

some of these comments have nothing to do with the original post. Im getting confused. Please stay on topic.

**Comment by imachug** [comment]
@JazzMaster YOU are wrong. Look at [zeronet-win-dist](https://github.com/HelloZeroNet/ZeroNet-win-dist).

**Comment by krixano** [comment]
@JazzMaster You just spouted a bunch of incorrect stuff, which is very annoying.

Firstly, zeronet-win-dist *does* use an executable. Why can this be? Simple. You can compile python, you can run a python script from an executable, etc. etc. How do you think compiled programs are able to use plugins written in python? Because you can call python from an executable.

Next, Unix uses `/`, and Windows *typically* uses `\`. However, Windows *does* also support `/` (although, I'm not sure if unix supports `\` - never tried it). Type `cd /Users` in the command prompt and you can clearly see that Windows supports `/`. You can of course also use `\`.

Finally, they were staying on topic. And if they weren't, it doesn't matter because both the person who created the issue and the person who owns this repository dictate what's acceptable to be talked about in this topic and what's not.

**Comment by martinvahi** [comment]
If revision 3703 that is published at the ZeroNet GitHub trunk on 2018_11_08 (commit comment: "Rev3703") is OK for You and You are willing to out-comment 3 lines as shown at the screen video with Estonian comments

https://technology.softf1.com/software_by_third_parties/ZeroNet/2019_11_24_initially_failing_and_Ctrl-c-killed_unprepared_Estonian_demo_t1_how_to_get_the_2018_11_08_ZeroNet_Rev3703_mmmv_repackaging_t1_to_run_on_Windows10_Linux_layer.webm

then You may use my repackaged version 

https://technology.softf1.com/software_by_third_parties/ZeroNet/

on Windows10 with the  "Windows Subsystem for Linux", 

https://docs.microsoft.com/en-us/windows/wsl/faq

Debian distribution. The ZeroNet code at my package is unmodified, You may just clone the original source from the ZeroNet GitHub repository, revert to the given version and recursively diff. What I did was just write 3 Bash scripts, the ones at the root folder and, well, if You read the 3 Bash scripts, You'll see it all. With some hick-ups it does work on Windows10, as shown at the screen video.
